### ISSUE TITLE
Angular2 CellRenderer: api.refreshCells not working

### ISSUE THREAD

**Issue by stephanjust** [ceolter/ag-grid-ng2#30]
api.refreshCells is not working for Angular2 CellRenderers. The culprit seems to be that RenderedCell.prototype.refreshCell checks, in case that there's no new data, if the CellRenderer implements refresh().
```
if (!newData && this.cellRenderer && this.cellRenderer.refresh) {
    // if the cell renderer has a refresh method, we call this instead of doing a refresh
    // note: should pass in params here instead of value?? so that client has formattedValue
    var valueFormatted = this.formatValue(this.value);
    var cellRendererParams = this.column.getColDef().cellRendererParams;
    var params = this.createRendererAndRefreshParams(valueFormatted, cellRendererParams);
    this.cellRenderer.refresh(params);
    // need to check rules. note, we ignore colDef classes and styles, these are assumed to be static
    this.addClassesFromRules();
}
```
The Angular2 CellRenderer is simply wrapped by a "normal" CellRenderer that implements refresh 
```
CellRenderer.prototype.refresh = function (params) {
    this._params = params;
};
```
This refresh obviously doesn't refresh the Angular component. Possible solutions: recreate the Angular component as if refresh wasn't implemented, or add refresh to the AgAware interface and let the component handle refreshing (I'd like this a lot) instead of the wrapper renderer..

Use case: apart from the imo obvious necessity of adding the refresh method to the angular component: I'm editing one cell in a row that changes another cell in the same row (meaning, cellA restricts and potentially changes values of cellB). After these changes cellA gets recreated, but cellB remains unchanged (but the values in my data rows are updated).

**Comment by ceolter** [comment]
thank you for the feedback. we are gong to review the ng2 component again next week and will take this on board.

**Comment by seanlandsman** [comment]
Hi

I've added a refresh method to AgAware - this will be invoked on a cellrenderer component if it exists.

The code is now: 

`            refresh(params:any):void {
                this._params = params;
                if(this._componentRef.instance.refresh) {
                    this._componentRef.instance.refresh(params);
                }
            }
`

I've also updated from-component.component.ts in the ag-grid-ng2-examples project to use this.

This should solve your usecase I think, but do let me know if you have feedback 

thanks!
